src/storm/storage/expressions/BinaryNumericalFunctionExpression.cpp

```diff
@@ -90,6 +106,9 @@ int_fast64_t BinaryNumericalFunctionExpression::evaluateAsInt(Valuation const* v
         case OperatorType::Modulo:
             result = firstOperandEvaluation % secondOperandEvaluation;
             break;
+        case OperatorType::Logarithm:
+            result = logHelper(firstOperandEvaluation, secondOperandEvaluation);
```

Comment:
Should this throw an exception when the result is not an integer?

Reply:
I think evaluateAsInt is just supposed to do integer arithmetic. For example we also do not throw for divide and power.

Reply:
Good to know. The [description of the function](https://github.com/moves-rwth/storm/blob/master/src/storm/storage/expressions/BaseExpression.h#L78) made it seem like it should throw an exception if the result is not integer (but casted to one).

Reply:
Yeah this is not very consequent. Afaik, the description refers to the 'type' as a syntactical thing. Like evaluating `0.5` (a RationalLiteralExpression) as int should throw an error while evaluating `1/2` (the division between two IntegerLiteralExpressions) works.